Lib/pathlib/_abc.py

```diff
@@ -583,6 +583,11 @@ def _ensure_distinct_path(self, other_path):
         """
         Raise OSError(EINVAL) if the other path is within this path.
         """
+        # Note: there is no straightforward, foolproof algorithm to determine
+        # if one directory is within another. A particularly perverse example:
+        # consider a single network share mounted in one location via NFS, and
+        # in another location via CIFS. This method simply checks whether the
+        # other_path is lexically equal to, or within, this path.
```

Comment:
Alternative wording:

```py
# Note: there is no straightforward, foolproof algorithm to determine
# if one directory is within another (a particularly perverse example
# would be a single network share mounted in one location via NFS, and
# in another location via CIFS), so we simply checks whether the
# other path is lexically equal to, or within, this path.
```

I'm considering putting the perverse example in parentheses because I thought (while reading) that you would tell me *how* you would deal with that example, thereby proving me that it would be futile to implement such algorithm.